core/src/main/java/com/linecorp/armeria/client/HttpChannelPool.java

```diff
@@ -373,14 +386,26 @@ private void connect(SessionProtocol desiredProtocol, PoolKey key, ChannelAcquis
      * A low-level operation that triggers a new connection attempt. Used only by:
      * <ul>
      *   <li>{@link #connect(SessionProtocol, PoolKey, ChannelAcquisitionFuture,
-     *       ClientConnectionTimingsBuilder)} - The pool has been exhausted.</li>
+     *       ClientConnectionTimingsBuilder, URI)} - The pool has been exhausted.</li>
      *   <li>{@link HttpSessionHandler} - HTTP/2 upgrade has failed.</li>
      * </ul>
      */
     void connect(SocketAddress remoteAddress, SessionProtocol desiredProtocol,
-                 Promise<Channel> sessionPromise) {
+                 Promise<Channel> sessionPromise, URI reqUri) {
```

Comment:
`reqUri` could come before the promise. (because it looks more organized? promise/future usually comes at the end (or the first) because it's a sort of an output parameter.)